server/src/app.ts

```diff
@@ -76,10 +77,10 @@ createConnection().then(async () => {
   app.use('/user', userRoute)
   app.use('/game', gameRouter)
   
-  io.once('connection', async (socket) => {
-    io.use(authSocketUser)
+  io.on('connection', async (socket: Socket) => {
+    io.use((socket: Socket, next: any) => authSocketUser(socket, io, next))
+    await socketEventHandler(socket, io)
```

Comment:
The reason for all the issues with multiple socket event happening was due to the fact that we where registering a new event handler on every request.